src/Utils.cc

```diff
@@ -73,17 +73,14 @@ void dumpSpirv(std::string_view Spirv) {
   SpirvFile.close();
 }
 
-/// Returns true if the file can be executed.
-static bool canExecute(const fs::path &Path) {
-  if (!fs::exists(Path))
+/// Returns true if the hipcc can be executed by the user.
+static bool canExecuteHipcc(const fs::path &HipccPath) {
+  if (!fs::exists(HipccPath))
     return false;
 
-  const auto &Perms = fs::status(Path).permissions();
-
-  // Only considering file to be executable if it has others_exec bit
-  // set on. For owner_exec and group_exec bits we need to check
-  // user's file access privileges.
-  return (Perms & fs::perms::others_exec) != fs::perms::none;
+  auto Cmd = std::string("sh -c '\"") + HipccPath.string() +
```

Comment:
Is this called often? If so, perhaps cache the return value in a static variable?

Reply:
Every time a HIP compilation occurs. I think the overhead is not significant but the caching is a small and easy thing to add so I added it.